core/src/main/scala/org/apache/spark/deploy/ApplicationDescription.scala

```diff
@@ -28,5 +28,14 @@ private[spark] class ApplicationDescription(
 
   val user = System.getProperty("user.name", "<unknown>")
 
+  def copy(
```

Comment:
Well... I was wondering why it isn't a case class, but it has a mutable field. I hope it will become a case class at some point, and then the copy method will be just removed, but I would not mix here such refactoring.